android-components/components/feature/privatemode/src/test/java/mozilla/components/feature/privatemode/notification/AbstractPrivateNotificationServiceTest.kt

```diff
@@ -60,7 +64,7 @@ class AbstractPrivateNotificationServiceTest {
     }
 
     @Test
-    fun `WHEN the service is created THEN start foreground is called`() {
+    fun `WHEN the service is created THEN start foreground is called`() = runTestOnMain {
```

Comment:
This `runTestOnMain` is using a different dispatcher (Main) from the one that is default in the `MockService.scope` (TestDispatcher), so I suspect we will have intermittently failing tests if we don't use the same one.

You can share a reference to one of them so that there is only one. Probably the main one is preferable since we're using that internally when we shift to calling `notify`.